### ISSUE TITLE
How to create a tag before build command

### ISSUE THREAD

**Issue by mike-pisman** [python-semantic-release/python-semantic-release#698]
Hi, I am using `setuptools-scm` for automatic versioning from git tag. This works well when the tag already exists.

I run semantic-release GH action, which creates tag, release, and release notes, but does not attack built package.
In `pyproject.toml` I use build command `python -m pip install build && python -m build .`  
However, it builds the distribution package using the old tag.

For example, my previous tag was **0.6.2**. The semantic release correctly identified the new version(**0.7.0**).

```
           INFO     [semantic_release.version.algorithm] INFO    algorithm.py:60
                    algorithm.tags_and_versions: found 3                        
                    previous tags                                               
The next version is: 0.7.0! 🚀
           INFO     [semantic_release.cli.commands.version] INFO  version.py:333
                    version.version: Running build command python               
                    -m pip install build && python -m build . 
 🛠 Running build command: python -m pip install build && python -m build .

```
Yet the package is built using old version **0.6.2**
```
creating unipoll-api-0.6.2.dev46+g1f14bab
```

The question is how can I create a tag with new version(**0.7.0**) before *build* command, so the `setuptools-scm` will pick it up.

I believe the fact that the package has different from release version, causes **semantic-release** to not upload the new package when publishing the release.
```
[08:47:18] WARNING  [semantic_release.hvcs.util] WARNING              util.py:75
                    util._wrapper: Github.get_release_id_by_tag                 
                    received response 404: 404 Client Error: Not                
                    Found for url:                                              
                    https://api.github.com/repos/unipoll/API/releases           
                    /tags/                                                      
           WARNING  [semantic_release.hvcs.github] WARNING         github.py:251
                    github.upload_dists: No release corresponds to              
                    tag , can't upload dists      
```

---

Is it possible to build a package separately from semantic-release action? 
For exampe:

```
- name: Python Semantic Release
  uses: python-semantic-release/python-semantic-release@master
  with:
      github_token: ${{ secrets.GITHUB_TOKEN }}
- name: Set up Python
  uses: actions/setup-python@v2
  with:
    python-version: '3.10'
- name: Install dependencies
  run: |
    python -m pip install build
- name: Build package
  run: |
    python -m build
- name: Publish package to GitHub Release
  uses: python-semantic-release/upload-to-gh-release@main
  if: ${{ steps.release.outputs.released }} == 'true'
  with:
    github_token: ${{ secrets.GITHUB_TOKEN }}
    tag: ${{ steps.release.outputs.tag }}
```

**Comment by mike-pisman** [comment]
So, the build issue seemed to resolve itself for now; However, there is a problem with empty `steps.release.outputs.tag` variable.

The end of the `python-semantic-release` action indicated that the output variable `steps.release.outputs.tag` has been set
```console
##[debug]Docker Action run completed with exit code 0
##[debug]Set output released = true
##[debug]Set output version = 0.7.1
##[debug]Set output tag = v0.7.1
##[debug]Finishing: Python Semantic Release
```

However, in the `upload-to-gh-release` it's `null`

```console
##[debug]Evaluating: steps.release.outputs.tag
##[debug]Evaluating Index:
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating steps:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'release'
##[debug]....=> null
##[debug]..=> null
##[debug]=> null
##[debug]Result: null
##[debug]Loading env
```


My action:
```yaml

jobs:
  release:
    runs-on: arc-runner-k8s
    concurrency: release
    permissions:
      id-token: write
      contents: write
    steps:
    - name: Force Install GIT latest
      run: |
        sudo apt-get update
        sudo apt-get install -y software-properties-common
        sudo add-apt-repository -y ppa:git-core/ppa
        sudo apt-get update
        sudo apt-get install -y git
    - uses: actions/checkout@v3
      with:
        fetch-depth: 0
    - name: Python Semantic Release
      uses: python-semantic-release/python-semantic-release@master
      with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          force: "patch"
          build_command: "python -m pip install build && python -m build"
    - name: Publish package to GitHub Release
      uses: python-semantic-release/upload-to-gh-release@main
      if: ${{ steps.release.outputs.released }} == 'true'
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
        tag: ${{ steps.release.outputs.tag }}
```

Any help?

**Comment by bernardcooke53** [comment]
Hi, glad the build issue is resolved - I think the issue is your conditional, I hit this before and it's a known problem (https://github.com/actions/runner/issues/1173)

Change
 
```yaml
if: ${{ steps.release.outputs.released }} == 'true'
```
to
```yaml
if: steps.release.outputs.released == 'true'
```
Annoyingly using `${{ }}` makes the condition always evaluate to `true` even if it's not. We don't set the `tag` output if nothing's released

**Comment by mike-pisman** [comment]
@bernardcooke53, thank you for the reply, I found that issue yesterday. However, if it's always evaluated to `true`, wouldn't that tag output get populated?

Anyway, I have tried to remove brackets, not the condition is evaluated to false, since it the output of `released` is also `null.
```true && (null == 'true'))```

I have also added some steps to debug all of the output variables, and they all are null. Without brackets GitHub does not substitute the variables:
```yaml
- name: Python Semantic Release
      uses: python-semantic-release/python-semantic-release@master
      with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          force: "patch"
    - name: Debug release
      run: |
        echo "Released: ${{ steps.release.outputs.released }}"
        echo "Version: ${{ steps.release.outputs.version }}"
        echo "Tag: ${{ steps.release.outputs.tag }}"
    - name: Debug release without Brackets
      run: |
        echo "Released: steps.release.outputs.released"
        echo "Version: steps.release.outputs.version"
        echo "Tag: steps.release.outputs.tag"
```

And the output is:
```console
##[debug]Evaluating condition for step: 'Debug release'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Debug release
##[debug]Loading inputs
##[debug]Evaluating: format('echo "Released: {0}"
##[debug]echo "Version: {1}"
##[debug]echo "Tag: {2}"
##[debug]', steps.release.outputs.released, steps.release.outputs.version, steps.release.outputs.tag)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> 'echo "Released: {0}"
##[debug]echo "Version: {1}"
##[debug]echo "Tag: {2}"
##[debug]'
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating Index:
##[debug]........Evaluating steps:
##[debug]........=> Object
##[debug]........Evaluating String:
##[debug]........=> 'release'
##[debug]......=> null
##[debug]....=> null
##[debug]..=> null
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating Index:
##[debug]........Evaluating steps:
##[debug]........=> Object
##[debug]........Evaluating String:
##[debug]........=> 'release'
##[debug]......=> null
##[debug]....=> null
##[debug]..=> null
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating Index:
##[debug]........Evaluating steps:
##[debug]........=> Object
##[debug]........Evaluating String:
##[debug]........=> 'release'
##[debug]......=> null
##[debug]....=> null
##[debug]..=> null
##[debug]=> 'echo "Released: "
##[debug]echo "Version: "
##[debug]echo "Tag: "
##[debug]'
##[debug]Result: 'echo "Released: "
##[debug]echo "Version: "
##[debug]echo "Tag: "
##[debug]'
##[debug]Loading env
Run echo "Released: "
  echo "Released: "
  echo "Version: "
  echo "Tag: "
  shell: /usr/bin/bash -e {0}
##[debug]/usr/bin/bash -e /home/runner/_work/_temp/f0791aae-f791-41ed-ac04-e88d4cb2ba54.sh
Released: 
Version: 
Tag: 
##[debug]Finishing: Debug release
```

Here is the run: https://github.com/unipoll/API/actions/runs/6215785582/job/16869019457

Basically, the action does not set the variables for some reason...

**Comment by mike-pisman** [comment]
@bernardcooke53, thank you again, I found the issue.

I didn't notice that you need to provide the `id` for the step, I though that was the name of the job. Now everything works.
For anyone in this situation, make sure you release step has the `id` field like this:
```yaml
- id: release
  name: Python Semantic Release
  uses: python-semantic-release/python-semantic-release@master
  with:
      github_token: ${{ secrets.GITHUB_TOKEN }}
```
